vector/vector.go

```diff
@@ -1,34 +1,73 @@
+// Package vector contains the core operation for sorting.
 package vector
 
-import "sort"
-
-type Vector interface {
-	// Get Access i-th element
-	Get(i int) interface{}
-	// PushBack Add item at the end
-	PushBack(value interface{}) error
-	// Less Returns wether v1 is smaller than v2
-	Less(v1, v2 interface{}) bool
-	// Dump Create a file and store the underluing data
-	Dump(filename string) error
-	// FrontShift Remove the first element
-	FrontShift()
-	// End Length of the Vector
-	End() int
-	// insert Insert elements at index i
-	insert(i int, value interface{}) error
-	// convertFromString Convert the line from the file to the expected underlying data
-	convertFromString(value string) (interface{}, error)
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// Vector holds a slice of Elements for sorting. The NewElement is called each
+// time a new item from the file is read or inserted into the slice. The Less
+// function should return true if the first element is lower than the second.
+type Vector struct {
```

Comment:
I really like this new Vector regarding performance and clarity. It would be even better if the underlying structure does not have to be a slice. This is not top priority, but maybe someone would like to have a linked list/other data structure.

The only thing I have in mind would require another interface for the field s

Reply:
I'm with you. Let's keep slimming to the minimum requirements and then we have to refactor. As mentioned I'm not happy about the `Element` and I think we can come up with a better solution that would satisfy this as well.